objects/src/notes/note_type.rs

```diff
@@ -28,27 +28,68 @@ pub enum NoteType {
     Public = PUBLIC,
 }
 
+// CONVERSIONS FROM NOTE TYPE
+// ================================================================================================
+
 impl From<NoteType> for Felt {
     fn from(id: NoteType) -> Self {
         Felt::new(id as u64)
     }
 }
 
-impl TryFrom<Felt> for NoteType {
+// CONVERSIONS INTO NOTE TYPE
+// ================================================================================================
+
+impl TryFrom<u8> for NoteType {
     type Error = NoteError;
 
-    fn try_from(value: Felt) -> Result<Self, Self::Error> {
-        let value = value.as_int();
-        let note_type: u8 = value.try_into().map_err(|_| NoteError::InvalidNoteTypeValue(value))?;
-        match note_type {
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
             OFF_CHAIN => Ok(NoteType::OffChain),
             ENCRYPTED => Ok(NoteType::Encrypted),
             PUBLIC => Ok(NoteType::Public),
-            _ => Err(NoteError::InvalidNoteTypeValue(value)),
+            _ => Err(NoteError::InvalidNoteTypeValue(value.into())),
         }
     }
 }
 
+impl TryFrom<u16> for NoteType {
```

Comment:
I don't think, this particular conversion is necessary. We can simply cast smaller types as `u64` and then use conversion from `u64`.

Reply:
True - but I think it doesn't complicate things here either. I simplified the u16/u32 conversions a bit - but I think they are worth keeping.